tests/system/alerts/test_alerts.py

```diff
@@ -284,16 +284,17 @@ def test_job_failure_alert_sliding_window(self):
         with pytest.raises(Exception):
             self.project.run_function(function_name)
 
-        # Wait for more than two minutes to simulate a delay that is slightly longer than the alert period
-        time.sleep(125)
+        # wait for the periodic monitor runs function to run as it may take up to the maximum events_generation_interval
+        # to detect the event.
+        time.sleep(mlconf.alerts.events_generation_interval)
+
+        # Wait for more than half minute to simulate a delay that is slightly longer than the alert period
+        time.sleep(32)
```

Comment:
Just sleep at once, and give it a bit more
```suggestion
        # wait for the periodic monitor runs function to run as it may take up to the maximum events_generation_interval
        # to detect the event + an extra 40s to simulate a delay that is slightly longer than the alert period
        time.sleep(mlconf.alerts.events_generation_interval + 40)
```